### ISSUE TITLE
Cockpit doesn't log out after idleTimeout 

### ISSUE THREAD

**Issue by emaayan** [cockpit-project/cockpit#17827]
### Explain what happens

1. leaving cockpit alone for more then 10-15 minutes doesn't seem to cause it to logout 
2. i've tried it with several machines and incognito as well. 

are there any special conditions where the idle comes into play? 

### Version of Cockpit

215

### Where is the problem in Cockpit?

Overview

### Server operating system version

Ubuntu 18.04.06

### What browsers are you using?

Chrome

**Comment by marusak** [comment]
Have you set up the timer in /etc/cockpit/cockpit.conf?

```
[Session]
IdleTimeout=15
```

**Comment by emaayan** [comment]
yes it's actually to 10 minutes, this also happens on cockpit 264 on ubuntu 20

[WebService]
LoginTo=false
[Session]
IdleTimeout=10

**Comment by emaayan** [comment]
> yes it's actually to 10 minutes, this also happens on cockpit 264 on ubuntu 20
> 
> [WebService] LoginTo=false [Session] IdleTimeout=10


is there a debug setting i need to see why it won't logout?

**Comment by marusak** [comment]
We have similar report in bugzilla (https://bugzilla.redhat.com/show_bug.cgi?id=2092566 this one, but it is mostly private). We have no good idea about this though.

Is your setup special in anything? Or do you just enable cockpit service and go to port 9090?
What is version of `cockpit-system` package on your system?
Do you see any logs in your browser console around 10 minutes of inactivity?

**Comment by emaayan** [comment]
> We have similar report in bugzilla (https://bugzilla.redhat.com/show_bug.cgi?id=2092566 this one, but it is mostly private). We have no good idea about this though.
> 
> Is your setup special in anything? Or do you just enable cockpit service and go to port 9090? What is version of `cockpit-system` package on your system? Do you see any logs in your browser console around 10 minutes of inactivity?

i do, btw does writing logs to the browser console aborts the inactivity ? 
the issue seems to be inconsistent somehow..
![image](https://user-images.githubusercontent.com/13434544/197167150-cce1e78e-f987-441e-beaf-d6df4469cf6d.png)

**Comment by marusak** [comment]
Do you use any plugins/add-ons in chrome/edge?

**Comment by emaayan** [comment]
> Do you use any plugins/add-ons in chrome/edge?

on chrome yes, but not on edge.

**Comment by emaayan** [comment]
hi... just to refresh, this also happens with chrome (incognito) and edge when using version :

cockpit 283-1~bpo22.04.1
cockpit-bridge 283-1~bpo22.04.1
cockpit-networkmanager 283-1~bpo22.04.1
cockpit-packagekit 283-1~bpo22.04.1
cockpit-pcp 283-1~bpo22.04.1
cockpit-sosreport 283-1~bpo22.04.1
cockpit-storaged 283-1~bpo22.04.1
cockpit-system 283-1~bpo22.04.1
cockpit-ws 283-1~bpo22.04.1

**Comment by martinpitt** [comment]
I created a version of the shell with lots of debug console logs, and tested it on my system. Please download [shell-debug.tar.gz](https://github.com/cockpit-project/cockpit/files/10833499/shell-debug.tar.gz)
and unpack it in /usr/local/cockpit/ or ~/.local/share/cockpit/ on the affected machine (you most probably need to create these directories first)

Then log in, open the browser console (Ctrl+Shift+K in firefox, Ctrl+Shift+J in chromium), and switch away from the cockpit tab, so that e.g. mousemove events don't reset the idle timer.

You should get some initial "XXX ...idle.." messages, and then every 5s some new message that increases the idle timer. All debug messages start with "XXX", so you can filter the log view easily.

Can you please do that experiment, and give me the console log output for that whole time? At some point there should be some surprising event, either something goes wrong in the initialization, or something keeps resetting the idle timer every ~ 10 minutes.

We probably need another round of "refine logging in a specific area", but let's first see where the problem is.

**Comment by emaayan** [comment]
so i tried taht on version 264~1 
without admin acccess i got:
`
updates.js:2 Tracer failed: "{"problem":"access-denied","exit_status":null,"exit_signal":null,"message":"Not permitted to perform this action."}", data: "undefined"

(anonymous) @ updates.js:2
overview.js:2 failed to poll tuned re
(anonymous) @ overview.js:2
overview.js:2 failed to poll tuned re
(anonymous) @ overview.js:2
overview.js:2 failed to poll tuned re
(anonymous) @ overview.js:2
`

after turning in admin access i get 

`
Tracer failed: "{"problem":null,"exit_status":1,"exit_signal":null,"message":"Traceback (most recent call last):\n  File \"<string>\", line 2, in <module>\nModuleNotFoundError: No module named 'tracer'"}", data: """"
(anonymous) @ updates.js:2
(anonymous) @ cockpit.js:1
(anonymous) @ cockpit.js:1
L @ cockpit.js:1
setTimeout (async)
A @ cockpit.js:1
M @ cockpit.js:1
U @ cockpit.js:1
P.e.reject @ cockpit.js:1
(anonymous) @ cockpit.js:1
(anonymous) @ cockpit.js:1
L @ cockpit.js:1
setTimeout (async)
A @ cockpit.js:1
M @ cockpit.js:1
U @ cockpit.js:1
P.e.reject @ cockpit.js:1
(anonymous) @ cockpit.js:1
s @ cockpit.js:1
value @ cockpit.js:1
h @ cockpit.js:1
g @ cockpit.js:1
(anonymous) @ cockpit.js:1
A.n.onmessage.e.dispatch_data @ cockpit.js:1
(anonymous) @ cockpit.js:1
postMessage (async)
(anonymous) @ index.js:2
A.n.onmessage.e.dispatch_data @ cockpit.js:1

`

**Comment by martinpitt** [comment]
@emaayan Hmm, you said you are using version 283. The shell that I built is most probably not compatible with the other cockpit packages, 264 is very old.

**Comment by emaayan** [comment]
i was afraid you might say that, our prod versions are  actually 264~1,  in
the past  tried briefly on 283 just to see if it was resolved,
let me try again. hang on..

**Comment by emaayan** [comment]
> @emaayan Hmm, you said you are using version 283. The shell that I built is most probably not compatible with the other cockpit packages, 264 is very old.
well.. tried with 284, still same thing ..
![image](https://user-images.githubusercontent.com/13434544/221910892-d4a9ae7d-ce3e-4ab6-b38d-c24bca25be2f.png)

**Comment by martinpitt** [comment]
That tracer message is unrelated (and actually fixed by now). The point  here was to debug the idle timeout, with the extra "XXX debug messages, but your logs don't show any of them. Is that still an issue for you with current versions?

Right now this isn't actionable, so without the debug logs we'll have to close this. Thanks!

**Comment by emaayan** [comment]
looks like the new versions of cockpit have resolved this (version 303,305)

**Comment by martinpitt** [comment]
Cool, thanks for reporting back!
